source/util/hex_float.h

```diff
@@ -975,6 +975,39 @@ std::istream& operator>>(std::istream& is, HexFloat<T, Traits>& value) {
     is.get();
   }
 
+  // Returns a new value formed from 'value' by setting 'bit' at the given
+  // bit 'position'.  If 'position' indexes a bit position beyond the top
+  // of the integer type, then return the original value.
+  auto set_bit = [](uint_type value, uint_type bit,
```

Comment:
It's been a while since I have looked at this code, but I don't remember it being lambda heavy. Would we prefer to keep (especially) this lambda as a free function as it doesn't have any captures anyway?